pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexTreeBuilder.kt

```diff
@@ -142,6 +177,174 @@ class ApexTreeBuilder(val sourceCode: String, val parserOptions: ApexParserOptio
             buildChildren(node, parent = this, exclude = { it in node.modifiers })
         }
 
+    /**
+     * Builds an [ASTArrayStoreExpression] or [ASTArrayLoadExpression] wrapper for the
+     * [ArrayExpression] node.
+     */
+    private fun buildArrayExpression(node: ArrayExpression) =
+        if ((node.parent as? AssignExpression)?.target == node) {
+            ASTArrayStoreExpression(node)
+        } else {
+            ASTArrayLoadExpression(node)
+        }
+            .apply { buildChildren(node, parent = this) }
+
+    /**
+     * Builds an [ASTBinaryExpression], [ASTBooleanExpression], or [ASTInstanceOfExpression] wrapper
+     * for the [BinaryExpression] node.
+     */
+    private fun buildBinaryExpression(node: BinaryExpression) =
+        when (node.op) {
+            BinaryExpression.Operator.INSTANCEOF -> ASTInstanceOfExpression(node)
+            BinaryExpression.Operator.GREATER_THAN_OR_EQUAL,
+            BinaryExpression.Operator.GREATER_THAN,
+            BinaryExpression.Operator.LESS_THAN,
+            BinaryExpression.Operator.LESS_THAN_OR_EQUAL,
+            BinaryExpression.Operator.EQUAL,
+            BinaryExpression.Operator.NOT_EQUAL,
+            BinaryExpression.Operator.ALTERNATIVE_NOT_EQUAL,
+            BinaryExpression.Operator.EXACTLY_EQUAL,
+            BinaryExpression.Operator.EXACTLY_NOT_EQUAL,
+            BinaryExpression.Operator.BITWISE_XOR,
```

Comment:
is a bitwise XOR a boolean expression? I would have expected this together with bitwise or; and here, I would have expected a logical xor...

Reply:
That is a mistake, thank you for pointing this out